resources/views/components/chirps/chirp-card.blade.php

```diff
@@ -0,0 +1,143 @@
+                                </svg>
+                        </button>
+                    </x-slot>
+                    <x-slot name="content">
+                        <x-dropdown-link :href="route('chirps.edit', $chirp)">
+                                {{ __('Edit') }}
+                        </x-dropdown-link>
+                        <form method="POST" action="{{ route('chirps.destroy', $chirp) }}">
+                            @csrf
+                            @method('delete')
+                            <x-dropdown-link :href="route('chirps.destroy', $chirp)" onclick="event.preventDefault(); this.closest('form').submit();">
+                                    {{ __('Delete') }}
+                            </x-dropdown-link>
+                        </form>
+                    </x-slot>
+                </x-dropdown>
+            @endif
+        </div>        
+        <p class="mt-4 text-lg text-gray-900">{{ $chirp->message }}</p>
+        <!-- Display existing comments for the chirp -->
+        <div id="{{ $chirp->id }}">
+            @foreach ($chirp->comments as $comment)
+                <div id="{{ $comment->id }}" class="comment mt-4 pl-1 max-w-prose flex items-center border border-blue-500 justify-between rounded">
+                    {{ $comment->comment }}
+                    @if(auth()->user() && auth()->user()->id === $comment->user_id)
+                        <form method="POST" action="{{ route('comments.destroy', $comment) }}" data-id="{{ $comment->id }}"class="delete-comment lg:flex lg:items-center">
+                            @csrf
+                            <input type="hidden" name="comment_id" value="{{ $comment->id }}">
+                            <input type="hidden" id="user_id" name="user_id" value="{{ Auth::user() }}">
+                            <button type="submit" class="deleteButton bg-red-500 text-white py-1 px-2 m-1 rounded hover:bg-red-800">Delete</button>
+                        </form>
+                    @endif 
+                </div>
+            @endforeach
+        </div>
+        <form method="POST" action="{{ route('comments.store') }}" class="post-comment mt-4 lg:flex lg:items-center" name="commentForm">
+            @csrf
+            <input type="hidden" name="chirp_id" value="{{ $chirp->id }}">
+            <textarea name="comment" maxlength="255" class="w-full lg:w-1/4 p-1 border border-blue-500 rounded" placeholder="Add a comment" required></textarea>
+            <button type="submit" class="mt-2 bg-blue-500 text-white py-2 px-4 rounded ml-2 hover:bg-blue-800">Comment</button>
+        </form>     
+    </div>
+</div>
+
+<script>
+    function deleteComments(){
+        let forms = document.querySelectorAll("form.delete-comment");
+        forms.forEach(deleteButtonForm => {
+            deleteButtonForm.addEventListener("submit", async (event) => {
+                event.preventDefault();
```

Comment:
refactor this into a function addDeleteButtonEventListener, which takes in an element and adds the event listener to it. then, your foreach loop here passes the element into the function to add the event listener, and when you add a new element you can call that function again

Reply:
Excellent idea!